docs/advanced/porting_guide.md

```diff
@@ -36,6 +36,52 @@ Add the target description to ```mbed-os\targets\targets.json```:
 
 See the [mbed Target Documentation](mbed_targets.md) for more details on what this definition means.
 
+### Coding requirements
+
+All code submitted as part of a port must compile with GCC ARM Embedded, ARM Compiler 5 and IAR EWARM. Flags that port code must compile with for each compiler are listed below. The profiles used to build mbed OS are subsets of each of these flags with exactly one optimization setting.
+
+#### GCC ARM Embedded
+
+Code in a port must be compatible with the `softfp` float ABI and compile without warnings or errors using GCC ARM Embedded with the following flags.
+
+flag                        | meaning
+----------------------------|---------------------------------------
+`-Wall`                     | Enable most warnings
+`-Wextra`                   | Enable extra warnings
+`-fno-exceptions`           | Disable exceptions
+`-funsigned-char`           | The `char` type is unsigned by default
+`-fomit-frame-pointer`      | Do not keep a frame pointer
+`-fno-rtti`                 | Disable run time type information (C++ only)
+`-Os`                       | Optimize for size
+`-O0`                       | Do not optimize
+`-std=gnu99`                | C uses the GNU99 standard 
+`-std=gnu++98`              | C++ uses the GNU++98 standard
+
+#### ARM Compiler 5
+
+Code submitted as part of a port must compile without warnings or errors using ARM Compiler 5 with the following flags.
```

Comment:
This says that the errors use ARM Compiler 5 and flags. Is that correct? If not, please change to "Code submitted for porting with ARM Compiler 5 must compile without warnings or errors and must use the following flags" or something similar.